src/ordinal.rs

```diff
@@ -107,9 +107,17 @@ impl Ordinal {
 
     let cycle_start_epoch = cycle_number * CYCLE_EPOCHS;
 
-    let cycle_progression = period_offset
-      .checked_sub(epoch_offset % PERIOD_BLOCKS)
-      .ok_or_else(|| anyhow!("Invalid relationship between epoch offset and period offset"))?;
```

Comment:
I think you wanted to test that the absolute difference between the `epoch_offset` and the `period_offset` was always a multiple of `336` (`Epoch::BLOCKS % PERIOD_BLOCKS`).

Valid: `0°0'336''1234'''`
Valid: `0°1680'0''1234'''`
Invalid: `0°1680'1''1234'''`
Invalid:  `0°1'672''1234'''`